docs/development/codeguide_emacs.rst

```diff
@@ -1,207 +1,67 @@
-*********************************************
- Emacs setup for following coding guidelines
-*********************************************
+*******************************************
+Emacs setup for following coding guidelines
+*******************************************
 
-.. _flymake: http://www.emacswiki.org/emacs/FlyMake
-.. _pyflakes: http://pypi.python.org/pypi/pyflakes
-.. _pycodestyle: http://pypi.python.org/pypi/pycodestyle
-.. include:: workflow/known_projects.inc
+.. _flycheck: http://www.flycheck.org/
+.. _flake8: http://flake8.pycqa.org/
 
-The Astropy coding guidelines are listed in :doc:`codeguide`. This
-document will describe some configuration options for Emacs, that will
-help in ensuring that Python code satisfies the guidelines. Emacs can
-be configured in several different ways. So instead of providing a drop
-in configuration file, only the individual configurations are presented
-below.
+The Astropy coding guidelines are listed in :doc:`codeguide`. Here, we describe
+how to configure Emacs to help ensure Python code satisfies the guidelines.
 
-For this setup we will need flymake_, pyflakes_ and the pycodestyle_ Python
-script, in addition to ``python-mode``.
+For this setup, we add to the standard ``python-mode`` using flycheck_ and the
+flake8_ python style checker.  For installation instructions, see their
+respective web sites (or install via your distribution; e.g., in Debian/Ubuntu,
+the packages are called ``elpa-flycheck`` and ``flake8``).
 
-Flymake comes with Emacs 23. The rest can be obtained from their websites,
-or can be installed using `pip`_.
+.. note:: Emacs can be configured in several different ways. So instead of
+          providing a drop in configuration file, only the individual
+          configurations are presented below.
 
-Global settings
-***************
+          The setup below is on purpose minimal.  In principle, it is possible
+          to use `Emacs for Python development
+          <https://realpython.com/blog/python/emacs-the-best-python-editor/>`_,
+          with, e.g., `elpy <https://elpy.readthedocs.io/>`_.
 
 No tabs
 =======
 
-This setting will cause all tabs to be replaced with spaces. The number
-of spaces to use is set in the :ref:`basic settings` section below.
+This setting will cause indentation to use spaces rather than tabs for all
+files.  For python files, indendation of 4 spaces will be used if the tab key
+is pressed.
 
 .. code-block:: scheme
 
   ;; Don't use TABS for indentations.
   (setq-default indent-tabs-mode nil)
 
-Maximum number of characters in a line
```

Comment:
This whole setting has no effect, since `python-mode` ignores this for all but docstrings. Personally, I think one is better off not enabling it.